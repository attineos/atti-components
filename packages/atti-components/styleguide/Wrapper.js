import React from 'react'

import ThemeProvider from '../src/components/ThemeProvider'
import { defaultTheme } from '../src/themes'

export default ({ children }) => <ThemeProvider otherTheme={defaultTheme}>{children}</ThemeProvider>
