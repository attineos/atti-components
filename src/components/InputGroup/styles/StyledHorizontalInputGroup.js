import styled from 'styled-components'

const StyledHorizontalInputGroup = styled.div`
  > * {
    width: auto;
  }

  > *:first-child {
    border-bottom-right-radius: 0;
    border-top-right-radius: 0;

    border-right: none;
  }

  > *:last-child {
    border-bottom-left-radius: 0;
    border-top-left-radius: 0;
  }

  > *:not(:first-child):not(:last-child) {
    border-radius: 0;

    border-right: none;
  }
`

export default StyledHorizontalInputGroup
