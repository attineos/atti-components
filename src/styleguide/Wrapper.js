import React from 'react'
import PropTypes from 'prop-types'
import { remove, get } from 'lodash'

import Cookies from './helpers/Cookies'
import themes from './config/themes'

import ThemeProvider from '../components/ThemeProvider'

const updateWrappers = []

export const getWrappersToUpdate = () => updateWrappers

class Wrapper extends React.Component {
  static propTypes = {
    /**
     * The child node.
     */
    children: PropTypes.node.isRequired,
  }

  state = {
    selectedTheme: null,
  }

  componentWillMount() {
    updateWrappers.push(this)

    const selectedTheme = Cookies.hasItem('selectedTheme')
      ? Cookies.getItem('selectedTheme')
      : 'default'

    this.state.selectedTheme = get(themes, selectedTheme, null)
  }

  componentWillUnmount() {
    remove(updateWrappers, item => item === this)
  }

  handleSetTheme = newTheme =>
    this.setState(
      {
        selectedTheme: newTheme,
      },
      () => this.forceUpdate(),
    )

  render() {
    const { children } = this.props
    const { selectedTheme } = this.state

    return (
      <ThemeProvider theme={selectedTheme} key={Math.random()}>
        {children}
      </ThemeProvider>
    )
  }
}

export default Wrapper
