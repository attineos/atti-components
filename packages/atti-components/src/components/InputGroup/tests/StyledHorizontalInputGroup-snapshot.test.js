/* This file was generated automatically, please do not write in it. */
import React from 'react';
import StyledHorizontalInputGroup from '../styles/StyledHorizontalInputGroup';
import Wrapper from 'styleguide/Wrapper'
import renderer from 'react-test-renderer';

const mock = () => {}

it('StyledHorizontalInputGroup renders correctly', () => {
  const tree = renderer
    .create(
      <Wrapper>
        <StyledHorizontalInputGroup ></StyledHorizontalInputGroup>
      </Wrapper>
      )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
    