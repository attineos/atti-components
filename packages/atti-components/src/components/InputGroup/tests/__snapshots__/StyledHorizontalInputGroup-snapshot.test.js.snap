// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledHorizontalInputGroup renders correctly 1`] = `
<span
  className="atti-box"
>
  <div
    className="StyledHorizontalInputGroup-sc-14ezq2l-0 bPFlwY"
  />
</span>
`;
